collector.go

```diff
@@ -118,6 +128,7 @@ func (c collector) Collect(ch chan<- prometheus.Metric) {
 	if err != nil {
 		log.Errorf("Error scraping target %s: %s", c.target, err)
 		ch <- prometheus.NewInvalidMetric(prometheus.NewDesc("snmp_error", "Error scraping target", nil, nil), err)
+		snmpCollectionErrors.Inc()
```

Comment:
Isn't this double-counting with ScrapeTarget? Rarely should one metric like this be incremented in more than one function.